src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBreakpointEvidenceSpark.java

```diff
@@ -0,0 +1,1160 @@
+                log("Ignoring " + kmerKillSet.size() + " genomically common kmers.");
+                final List<SVKmer> kmerKillList = getHighCountKmers(reads);
+                log("Ignoring " + kmerKillList.size() + " common kmers in the reads.");
+                kmerKillSet.addAll(kmerKillList);
+                log("Ignoring a total of " + kmerKillSet.size() + " unique common kmers.");
+
+                kmerAndIntervalsSet = new HopscotchHashSet<>(
+                        getKmerIntervals(ctx, qNamesSet, kmerKillSet, reads, locations, pipelineOptions));
+                log("Discovered " + kmerAndIntervalsSet.size() + " kmers.");
+            }
+
+            qNamesSet.addAll(getAssemblyQNames(ctx, kmerAndIntervalsSet, reads));
+
+            if ( locations.qNamesAssemblyFile != null ) {
+                dumpQNames(locations.qNamesAssemblyFile, pipelineOptions, qNamesSet);
+            }
+
+            log("Discovered "+qNamesSet.size()+" unique template names for assembly.");
+        }
+        final String outDir = outputDir;
+        generateFastqs(ctx, qNamesSet, nIntervals, reads,
+                intervalAndFastqBytes -> writeFastq(intervalAndFastqBytes, outDir));
+        log("Wrote FASTQs for assembly.");
+    }
+
+    @VisibleForTesting static void generateFastqs(final JavaSparkContext ctx,
+                                       final HopscotchHashSet<QNameAndInterval> qNamesSet,
+                                       final int nIntervals,
+                                       final JavaRDD<GATKRead> reads,
+                                       final VoidFunction<Tuple2<Integer, List<byte[]>>> fastqWriter) {
+        final Broadcast<HopscotchHashSet<QNameAndInterval>> broadcastQNamesSet = ctx.broadcast(qNamesSet);
+
+        reads
+            .filter(read -> !read.isSecondaryAlignment() && !read.isSupplementaryAlignment())
+            .mapPartitionsToPair(readItr ->
+                    new ReadsForQNamesFinder(broadcastQNamesSet.value(), nIntervals).call(readItr), false)
+            .reduceByKey(FindBreakpointEvidenceSpark::combineLists)
+            .foreach(fastqWriter);
+
+        broadcastQNamesSet.destroy();
+    }
+
+    private static List<byte[]> combineLists( final List<byte[]> list1, final List<byte[]> list2 ) {
+        final List<byte[]> result = new ArrayList<>(list1.size() + list2.size());
+        result.addAll(list1);
+        result.addAll(list2);
+        return result;
+    }
+
+    @VisibleForTesting static byte[][] sortFastqRecs( final List<byte[]> fastqRecs ) {
+        final int nFastqs = SVUtils.iteratorSize(fastqRecs.iterator());
+        final byte[][] fastqsArray = new byte[nFastqs][];
+
+        final Iterator<byte[]> itr = fastqRecs.iterator();
+        for ( int idx = 0; idx != nFastqs; ++idx ) {
+            if ( !itr.hasNext() ) throw new GATKException("fastq iterator crapped out at element "+idx);
```

Comment:
I know we should never hit this but probably better not to put 'crapped' in a message that could bubble out to the user. Actually, could we just get rid of the iterator and do `fastqRecs.get(i)`? Or will this be a LinkedList?